src/decideid_assets/src/components/auth/new_profile/NewProfile.tsx

```diff
@@ -1,18 +1,34 @@
 import React, { useState } from "react";
 import { useProfile } from "../../../utils";
+import { useNavigate } from "react-router-dom";
+import { useEffect } from "react";
 
 export default function NewProfile() {
+  const navigate = useNavigate();
+  const { createProfile, profile, isLoading } = useProfile();
```

Comment:
If we are using global StateManagement on FE - why do we need such hook ? I believe we can have just `init()` method which preloads minimal bunch of necessary data...